x-pack/platform/plugins/shared/search_connectors/server/plugin.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  PluginInitializerContext,
+  Plugin,
+  CoreSetup,
+  CoreStart,
+  LoggerFactory,
+} from '@kbn/core/server';
+import type { ConnectorServerSideDefinition } from '@kbn/search-connectors';
+import { getConnectorTypes } from '../common/lib/connector_types';
+import type {
+  SearchConnectorsPluginSetup,
+  SearchConnectorsPluginStart,
+  SearchConnectorsPluginSetupDependencies,
+  SearchConnectorsPluginStartDependencies,
+} from './types';
+import { registerConnectorRoutes } from './routes/connectors';
+import { registerStatsRoutes } from './routes/stats';
+import { registerMappingRoute } from './routes/mapping';
+import { registerSearchRoute } from './routes/search';
+import { PLUGIN_ID } from '../common/constants';
+import { registerApiKeysRoutes } from './routes/api_keys';
+
+export class SearchConnectorsPlugin
+  implements
+    Plugin<
+      SearchConnectorsPluginSetup,
+      SearchConnectorsPluginStart,
+      SearchConnectorsPluginSetupDependencies,
+      SearchConnectorsPluginStartDependencies
+    >
+{
+  private connectors: ConnectorServerSideDefinition[];
+  private readonly logger: LoggerFactory;
+
+  constructor(initializerContext: PluginInitializerContext) {
+    this.connectors = [];
+    this.logger = initializerContext.logger;
+  }
+
+  public setup(
+    coreSetup: CoreSetup<SearchConnectorsPluginStartDependencies, SearchConnectorsPluginStart>,
+    plugins: SearchConnectorsPluginSetupDependencies
+  ) {
+    const http = coreSetup.http;
+
+    plugins.features.registerElasticsearchFeature({
+      id: PLUGIN_ID,
+      management: {
+        data: [PLUGIN_ID],
+      },
+      privileges: [
+        {
+          requiredClusterPrivileges: ['monitor'],
+          ui: [],
+        },
+      ],
+    });
+
+    this.connectors = getConnectorTypes(http.staticAssets);
+    const router = http.createRouter();
+
+    // Enterprise Search Routes
+    if (this.connectors.length > 0) {
```

Comment:
if this is 0 should we register any routes? should the whole plugin and UI be enabled? When would this be 0?

Reply:
Moved all routes under the if check https://github.com/elastic/kibana/pull/213509/commits/212d6bdde342e7a5fa9a7d1e0a7b726d4b39f5aa. But before the changes I made plugin was enabled even the number of connector types was 0. I may change this logic if you think it is reasonable.